pkg/metrics/collector/pod_ip_metrics.go

```diff
@@ -318,26 +334,104 @@ func (c *podIPMetricsCollector) setupDirectoryWatcher(dir string) error {
 	if err != nil {
 		glog.Errorf("Failed to add watcher for directory %s: %v", dir, err)
 	}
-	podIPMetricsWatcherSetup = true
+	podIPMetricsWatcherIsIntialized = true
+	return nil
+}
+
+func (c *podIPMetricsCollector) countIPsFromRange(subnet types.IPNet) (uint64, error) {
+	ones, bits := subnet.Mask.Size()
+	if bits <= ones {
+		return 0, fmt.Errorf("invalid subnet mask: %v", subnet.Mask)
+	}
+	// this checks that we are not overflowing an int64
+	if bits-ones >= 64 {
+		return math.MaxUint64, nil
+	}
+	max := uint64(1) << uint(bits-ones)
+	// Don't use the network's ".0" address,
+	if max == 0 {
+		return 0, fmt.Errorf("subnet includes only the network address")
+	}
+	max--
+	if strings.Contains(subnet.IP.String(), ".") {
+		// Don't use the IPv4 network's broadcast address
+		if max == 0 {
+			return 0, fmt.Errorf("subnet includes only the network and broadcast addresses")
+		}
+		max--
+	}
+	return max, nil
+}
+
+func (c *podIPMetricsCollector) updateAssignedIPs(subnet types.IPNet, totalIP uint64) {
+	if subnet.IP.To16() != nil && subnet.IP.To4() == nil {
+		c.assignedIPv6AddrCount += totalIP
+	} else if subnet.IP.To4() != nil {
+		c.assignedIPv4AddrCount += totalIP
+	}
+}
+
+func (c *podIPMetricsCollector) calculateAssignedIP(subnetFile string) error {
+	networkConfigList, err := libcni.ConfListFromFile(subnetFile)
+	if err != nil {
+		glog.Errorf("Error reading network config from file %s: %v", subnetFile, err)
+		return err
+	}
+
+	var ipamConfig *allocator.IPAMConfig
+	gkePluginFound := false
+	for _, plugin := range networkConfigList.Plugins {
+		if plugin.Network.Type == "gke" {
+			gkePluginFound = true
+			ipamConfig, _, err = allocator.LoadIPAMConfig(plugin.Bytes, "")
+			if err != nil {
+				glog.Errorf("Error loading IPAM config: %v", err)
+				return err
+			}
+			if len(ipamConfig.Ranges) == 0 {
+				return fmt.Errorf("no subnets assigned in the IPAM config")
+			}
+			break
+		}
+	}
+	if !gkePluginFound {
+		return fmt.Errorf("gke plugin not found in network config")
+	}
+
+	for _, ranges := range ipamConfig.Ranges {
+		for _, ipRange := range ranges {
+			totalIP, err := c.countIPsFromRange(ipRange.Subnet)
+			if err != nil {
+				glog.Errorf("Error calculating total IPs for subnet %s: %v", ipRange.Subnet.IP.String(), err)
+				return err
+			}
+			c.updateAssignedIPs(ipRange.Subnet, totalIP)
+		}
+	}
 	return nil
 }
 
 func (c *podIPMetricsCollector) Update(ch chan<- prometheus.Metric) error {
-	if !podIPMetricsWatcherSetup {
+	if !podIPMetricsWatcherIsIntialized {
 		if err := c.setupDirectoryWatcher(gkePodNetworkDir); err != nil {
 			glog.Errorf("setupDirectoryWatcher returned error: %v", err)
 			return nil
 		}
+		//calculate assigned IP count
```

Comment:
Doesn't look like this comment provides any value :)

Reply:
Done.